File: 2022-2023/sourcecode/lab/lab9/RoomRelationsJava/app/src/main/java/pl/edu/uwr/pum/roomrelationsjava/model/entities/Student.java
package pl.edu.uwr.pum.roomrelationsjava.model.entities;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity
public class Student {

    @NonNull
    @PrimaryKey
    private final String studentName;
    private final int indexNumber;
    private final String facultyName;

    public Student(@NonNull String studentName, int indexNumber, String facultyName) {
        this.studentName = studentName;
        this.indexNumber = indexNumber;
        this.facultyName = facultyName;
    }

    @NonNull
    public String getStudentName() {
        return studentName;
    }

    public int getIndexNumber() {
        return indexNumber;
    }

    public String getFacultyName() {
        return facultyName;
    }
}

File: 2022-2023/sourcecode/lab/lab9/RoomRelationsKotlin/app/src/main/java/pl/edu/uwr/pum/roomrelationskotlin/model/entity/Student.kt
package pl.edu.uwr.pum.roomrelationskotlin.model.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
data class Student (
    @PrimaryKey(autoGenerate = false)
    val studentName: String,
    val indexNumber: Int,
    val facultyName: String
        )